### PR TITLE
Tests: fix cached point parse contamination between tests

### PR DESCRIPTION
**Pull request by MetRonnie**
In back-compat mode the cycle point time zone is assumed to be local, whereas in normal mode it is assumed to be UTC. There was contamination of the point parse caching where the time zone would carry over from tests of back-compat vs normal mode.

Also removed pytest-env because it doesn't work with pytest-xdist (copied from master - see https://github.com/cylc/cylc-flow/pull/5826/commits/be46a11115294ab41f00b3a6a76ffacb29b28d32)

### CODE REVIEW

File: .github/workflows/test_fast.yml [hunk 1]
```diff
@@ -20,11 +20,17 @@ jobs:
       fail-fast: false # Don't let a failed MacOS run stop the Ubuntu runs
       matrix:
         os: ['ubuntu-latest']
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.7', '3.8', '3.10', '3.11']
```
Comment (hjoliver) [inline .github/workflows/test_fast.yml:23]: Is that change intentional?

Comment (MetRonnie) [reply .github/workflows/test_fast.yml:23]: Yes, so we test non-UTC on one of the python versions

**Review by hjoliver — APPROVED** [review]
